src/rules/trailingCommaRule.ts

```diff
@@ -20,11 +20,98 @@ import * as ts from "typescript";
 
 import * as Lint from "../index";
 
+type OptionValue = "always" | "never" | "ignore";
+
+interface CustomOptionValue {
+    arrays?: OptionValue;
+    exports?: OptionValue;
+    functions?: OptionValue;
+    imports?: OptionValue;
+    objects?: OptionValue;
+    typeLiterals?: OptionValue;
+}
+
 interface Options {
-    multiline?: "always" | "never";
-    singleline?: "always" | "never";
+    multiline?: OptionValue | CustomOptionValue;
```

Comment:
Yeah, I wasn't sure about this. Since I'm assigning the string value to the object keys in `normalizeOptions` I assumed it would make the type checker think that I was assigning e.g. `options.multiline.arrays` a `CustomOptionValue` if `options.multiline` is only declared as a `CustomOptionValue`. But this doesn't seem to cause any issues.